presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1548,6 +1557,17 @@ public HiveInsertTableHandle beginInsert(ConnectorSession session, ConnectorTabl
         return result;
     }
 
+    private HiveCompressionCodec getHiveCompressionCodec(ConnectorSession session, boolean isTemporaryTable, HiveStorageFormat storageFormat)
+    {
+        if (isTemporaryTable) {
+            return getTemporaryTableCompressionCodec(session);
+        }
+        if (storageFormat.equals(ORC) || storageFormat.equals(DWRF)) {
```

Comment:
- For the purpose of checking compatibility between `StorageFormat` and `HiveCompressionCodec`, you can use `HiveCompressionCodec::isSupportedStorageFormat`

- For the purpose of getting ORC specific compression method,  you can use `HiveCompressionCodec::orcCompressionKind``

Reply:
@viczhang861 We are already using copressionKind and isSupportedStorageFormat.

The reason to add an extra configuration was the unavailability of uniform support of all compression codec across all storage formats. The ZSTD compression codec is only available for orc and dwrf storage format.
We wanted to roll out the ZSTD configuration for orc/dwrf storage formats. And without a dedicated compression codec for these storage formats, we would have enabled this compression for all storage format. Which would have failed the pipeline as ZSTD is not supported for other storage formats.

Reply:
I see, another way to do this is to add a default compression codec into `HiveStorageFomat`, to avoid configuring all combinations between storage format and compression.

Just curious,  what are other storage formats in production that rely on `hive.compression-codec`  ?

Reply:
hive.compression-codec is used to configure compression for all storage format. I am not sure how many clients are using this config. 
Hardcoding default in the HiveStorageFormat would not be flexible. Using the configurations we can rollout the change as we need.